pkg/cacheutil/memcached_client.go

```diff
@@ -250,6 +264,12 @@ func (c *memcachedClient) Stop() {
 }
 
 func (c *memcachedClient) SetAsync(ctx context.Context, key string, value []byte, ttl time.Duration) (err error) {
+	// Skip hitting memcached at all if the item is bigger than the max allowed size.
+	if c.config.MaxItemSize > 0 && len(value) > int(c.config.MaxItemSize) {
```

Comment:
`int(c.config.MaxItemSize)` seems unsafe as it can cut off higher 4 bytes of the value on 32-bit systems. On the other hand, MaxItemSize is very unlikely to be bigger than 2^31, so not worth fixing.

Reply:
Unless I'm missing anything, a slice in Go is limited to an `int` size, so I'm not sure `len(value)` could be ever greater than `int`. Am I missing anything?

Reply:
From an offline discussion:

> It's c.config.MaxItemSize that can be bigger than 2^31. For example, say that c.config.MaxItemSize = 2^31 + 10, then conversion to int will make it negative number

You're right. Fixed!